### DISCUSSION TITLE
Concurrency: cancel-in-progress at top level doesn't seem to cancel running workflows?

### DISCUSSION THREAD

**Discussion by ocean90** [community/community#26566] [Actions]
<p>Hi there, I just tried the new shipped concurrency feature (thanks for that!) and with the following workflow in a private repo:</p>
<pre><code class="lang-yaml">name: Deploy

on:
  push:
    branches:
      - master
      - production
    paths-ignore:
      - '**.md'

# Ensures that only one deploy task per branch/environment will run at a time.
concurrency:
  group: environment-${{ github.ref }}
  cancel-in-progress: true

jobs:

  deploy:
    name: Deploy
    runs-on: ubuntu-latest

    steps:
       # All the steps
</code></pre>
<p>Then I did two pushes, both triggered the workflow but the second one didn’t cancel the first, instead it was pending and waiting for the first one to finish.<br>
From the documentation it’s not clear, whether <code>cancel-in-progress</code> is actually supported on the top-level  <code>concurrency</code> key. Both sections (<a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#concurrency" rel="noopener nofollow ugc">this</a> and <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idconcurrency" rel="noopener nofollow ugc">this</a>) only mention  “in-progress job or run” and not “workflow”. Does somebody know whether this is supported or not? Maybe this is some sort of a copy/paste error?</p>
<p>Thanks!</p>

**Comment by Simran-B** [comment]
<p>The first link is about the <code>concurrency</code> key at the top level:</p>
<blockquote>
<p>To also cancel any currently running job or workflow in the same concurrency group, specify <code>cancel-in-progress: true</code>.</p>
</blockquote>
<p>It sounds like it should cancel pending as well as running workflows with the same group. If it doesn’t work then that might be a bug (remember that the feature is in beta).</p>

**Comment by ocean90** [comment]
<p>So far I was only able to get it to work when using <code>cancel-in-progress</code> at the job level.</p>

**Comment by Simran-B** [comment]
<p>There’s a remark about <code>jobs.&lt;job_id&gt;.concurrency</code>:</p>
<blockquote>
<p>Note: When concurrency is specified at the job level, order is not guaranteed for jobs or runs that queue within 5 minutes of each other.</p>
</blockquote>
<p>I wonder if this is related to your issue. Does it mean that a later queued job or workflow run may get canceled instead of the earlier queued one? But in your case neither is canceled…</p>

**Comment by eseay** [comment]
<p><a class="mention" href="/ocean90">@ocean90</a> I was very eager to utilize this feature when I saw the announcement, but I have had the same experience as you.</p>
<p>I inferred from the documentation that <code>concurrency</code> can be specified at the top level  of the workflow just as you have (as well as at the job level, which I don’t presently have a use for).</p>
<p>I am using <code>${{ github.workflow }}-${{ github.ref }}</code> as my concurrency key because I have multiple workflows that respond to the <code>push</code> event on some of these branches.</p>
<pre><code class="lang-auto">name: Create Alpha Build
on:
  push:
    branches: 
      - develop
      - release/*
       - '!release/**-rc*'
     tags: v2021.*

concurrency: 
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true

# .....
</code></pre>

**Comment by ocean90** [comment]
<aside class="quote no-group" data-username="Simran-B" data-post="4" data-topic="175801">
<div class="title">
<div class="quote-controls"></div>
 Simran-B:</div>
<blockquote>
<p>I wonder if this is related to your issue.</p>
</blockquote>
</aside>
<p>This should only apply if you use the key on the job level which I didn’t when I reported the issue. I wasn’t able to trigger this issue now with using <code>concurrency</code> on the job level either though.</p>
<p>Another bug was mentioned in <a href="https://github.community/t/actions-concurrency-bug-report/175695" class="inline-onebox">Actions concurrency bug report</a> but I’m not sure if they are related, maybe <a class="mention" href="/yaananth">@yaananth</a> can answer that?</p>

**Comment by ocean90** [comment, marked as answer]
<p><a class="mention" href="/yaananth">@yaananth</a> gave an update in <a href="https://github.community/t/actions-concurrency-bug-report/175695/4" class="inline-onebox">Actions concurrency bug report - #4 by yaananth</a> for a related issue and so I tried the <code>concurrency</code> key on the workflow level again and it’s now working too!</p>
